Src/IronPython.Modules/_ctypes/SimpleType.cs

```diff
@@ -84,12 +86,42 @@ public SimpleType(CodeContext/*!*/ context, string name, PythonTuple bases, Pyth
                     default:
                         throw new NotImplementedException("simple type " + sVal);
                 }
+
+                if (!name.EndsWith("_be") && !name.EndsWith("_le") && swappedTypes.IndexOf(_charType) != -1) {
+                    CreateSwappedType(context, name, bases, dict);
+                }
+                _format = (BitConverter.IsLittleEndian ? '<' : '>') + _charType.ToString();
+                _getValue = GetVal;
+                _setValue = SetVal;
             }
 
             private SimpleType(Type underlyingSystemType)
                 : base(underlyingSystemType) {
             }
 
+            private SimpleType(CodeContext/*!*/ context, string name, PythonTuple bases, PythonDictionary dict, bool isLittleEndian) : this(context, name, bases, dict) {
+                _format = (isLittleEndian ? '<' : '>') + _charType.ToString();
+                if(isLittleEndian != BitConverter.IsLittleEndian) {
```

Comment:
I wonder if it would be better to simply have a `swap` field and move down the swapping to the `MemoryHolder.Read/Write` methods. It would avoid using delegates and having the mostly duplicated `Get/SetValue` methods.